pkg/git/repository.go

```diff
@@ -312,14 +319,44 @@ func (r *repo) commit(opts *PushOptions) (*plumbing.Hash, error) {
 		}
 	}
 
-	h, err = w.Commit(opts.CommitMsg, &gg.CommitOptions{All: true})
+	h, err = w.Commit(opts.CommitMsg, &gg.CommitOptions{
+		All:    true,
+		Author: author,
+	})
 	if err != nil {
 		return nil, err
 	}
 
 	return &h, nil
 }
 
+func (r *repo) getAuthor(ctx context.Context, providerType string) (*object.Signature, error) {
+	username, email, err := r.provider.GetAuthor(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get commiter data. Error: %w", err)
+	}
+
+	if username == "" || email == "" {
+		cfg, err := r.ConfigScoped(config.SystemScope)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get gitconfig. Error: %w", err)
+		}
+
+		username = cfg.User.Name
+		email = cfg.User.Email
+	}
+
+	if username == "" || email == "" {
+		return nil, fmt.Errorf("failed to get author data. Please make sure your gitconfig contains a name and an email")
```

Comment:
"failed to get author data" -> "missing required author information in git config, make sure your git config contains a 'user.name' and 'user.email'"